src/compiler/types.ts

```diff
@@ -1750,6 +1750,7 @@ namespace ts {
 
     export interface ReturnStatement extends Statement {
         kind: SyntaxKind.ReturnStatement;
+        returnKeyword: Token<SyntaxKind.ReturnKeyword>;

```

Comment:
I'd rather not increase the size of the node. Instead, consider modifying `writeTokenText` in emitter to emit comments as well as source maps. 

Keep in mind that `writeTokenText` is used quite a bit throughout the emitter. Changes could have an impact on emit performance. If the impact is minimal that change would be preferable. If it is significant then we either need to find ways to mitigate the cost or consider the approach in the current PR.